src/AutoUI/Filters/Filters.tsx

```diff
@@ -96,6 +106,159 @@ const removeFieldsWithNoFilter = (schema: JSONSchema): JSONSchema => {
 	return schema;
 };
 
+/**
+ * Constructs properties for schemas based on their x-ref-scheme or x-foreign-key-scheme annotations.
+ */
+const constructSchemaProperties = (
+	schemaOrProperties: JSONSchemaDefinition,
+	transformedXRefScheme: string[] | undefined,
+	description?: string,
+	title?: string,
+): JSONSchemaDefinition => {
+	if (schemaOrProperties == null || !transformedXRefScheme?.length) {
+		return schemaOrProperties;
+	}
+
+	const [firstRefSchemeKey, ...restRefScheme] = transformedXRefScheme;
+	if (isJSONSchema(schemaOrProperties) && schemaOrProperties.properties) {
+		return {
+			...schemaOrProperties,
+			title: title ?? schemaOrProperties.title,
+			description,
+			properties: constructSchemaProperties(
+				schemaOrProperties.properties,
+				restRefScheme,
+				description,
+			) as JSONSchema['properties'],
+		};
+	}
+	if (isJSONSchema(schemaOrProperties) && schemaOrProperties.items) {
+		return {
+			...schemaOrProperties,
+			title: title ?? schemaOrProperties.title,
+			description,
+			items: constructSchemaProperties(
+				schemaOrProperties.items as JSONSchema,
+				restRefScheme,
+				description,
+			),
+		};
+	}
+	if (firstRefSchemeKey) {
```

Comment:
A bit confused about how this bit works. Why do we split off the first key? Why do we only check it *after* the ifs to return depending on `schemaOrProperties.items` and `.properties`? Why do those two ifs need all but the first key passed into `constructSchemaProperties`?

(If you could add some comments to the code about this, that would be appreciated 🙏

Reply:
sure added some comments, in few words, we check if we ended the refScheme path, if so we reached the objective (probably a natural value type string|boolean|number), otherwise we keep doing recursive operations and we pass the restRefScheme which does not include this step as we just did it.

Reply:
Ah okay, thank you